contrib/epee/include/net/abstract_tcp_server2.inl

```diff
@@ -364,6 +364,7 @@ PRAGMA_WARNING_DISABLE_VS(4355)
       context.m_last_recv = time(NULL);
       context.m_recv_cnt += bytes_transferred;
       m_ready_to_close = false;
+      MDUMP(buffer_.data(), bytes_transferred, epee::net_utils::print_connection_context_short(context) + " <");
```

Comment:
The logging system has to acquire a lock and then scan/search a string before it can determine that the message is going to be skipped. So this has the effect of serializing all TCP read/writes with _all_ log messages, slowing things further than they are already are due to the debug/trace messages sprayed everywhere.

And the benefit is that we get a dump of arbitrary TCP data ... I don't see how this is a net gain. Some of the logs are going to be massive dumps.

At least drop some of the other log messages in these functions, the write function below in particular now has basically wasted trace log.

Reply:
I have a patch that does away with the lock actually.
And it's a gain when debugging.

Reply:
https://github.com/monero-project/monero/pull/5735

Reply:
That patch does not do away with the lock, it only skips the lock in some contexts. There are still `INFO` log statements that have to go through the full lock + filtering frequently since the default settings have `INFO` logging in _some_ categories. Removing the lock in all scenarios requires "fixed" (enum) categories (I think), so that the `set_level` call would compute the levels once "up-front" instead of "once-per-log-call".

This is dumping all P2P and RPC traffic in a series of text walls, right? Do you actually find that large information dump useful? The `DEBUG` logs already spew so much data that I typically avoid using it. I'll look at the patch again later ...

Reply:
The patch does not bypass the lock all the time, yes. It does bypass it for trace stuff (like this one in this patch) unless you've set things up to have at least one trace active, but in that case a lock isn't going to be your problem speedwise. I do not intend to remove the lock altogether.
This is dumping all network traffic, yes. I found this useful. It is not intended to be enabled unless you know you want to check what's going on in traffic (ie, timings of what's exchanged, what is exchanged, etc).

Reply:
So you're still objecting to this after 5735 ? If so I can keep it local I guess.

Reply:
The information in the net/p2p log is already high enough that I remove some of them, change them to `cout` as needed or just add my own. This would be something to add only if you think it helps with bug reports from users.